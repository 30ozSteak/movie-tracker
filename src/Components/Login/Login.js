import React, { Component } from 'react'

import { connect } from 'react-redux'
import { Route, NavLink } from 'react-router-dom'
import { loginUser } from '../../Actions/TheActionMan'
import HeaderLogin from './HeaderLogin/HeaderLogin'

class Login extends Component {
  constructor(props) {
    super(props)
    this.state = {
      // email: 'tman2272@aol.com',
      email: '',
      username: '',
      formState: '',
      password: '',
      // password: 'password',
    }
  }

  async componentDidMount() { }

  toggleActive = () => {
    this.state.formState === ""
      ? this.setState({ formState: "active" })
      : this.setState({ formState: "" });
  };

  resetForm = () => {
    this.setState({ password: '' })
  }

  handleChange = event => {
    const { value, name } = event.target
    this.setState({ [name]: value })
  }

  submitLogin = e => {
    const { loginCheck } = this.props
    const { email, password } = this.state
    e.preventDefault()
    loginCheck(email, password)
    this.resetForm()
  }

  render() {
    const { email, username, password } = this.state
    return (
      <section className="login-main">
        <HeaderLogin />
        {/* <nav>
          <NavLink to="/">Home</NavLink>
        </nav> */}
        <div className="user-image"></div>
        <section className='form-container'>
          <section className={this.state.formState}>
            <form
              className="login-form"
              onSubmit={this.submitLogin}
              aria-label="Existing user login form"
            >
              <p>Login</p>
              <h2>Welcome Back</h2>
              <label className="email-input">
                <input
                  tabIndex="0"
                  name="email"
                  placeholder="Email"
                  type="email"
                  value={email}
                  onChange={this.handleChange}
                  className="email-text"
                />
              </label>
              <hr />
              <label>
                <input
                  tabIndex="0"
                  name="password"
                  placeholder="Password"
                  type="password"
                  value={password}
                  onChange={this.handleChange}
                  className='password-text'
                />
              </label>
              <button className='submit-button'>Submit</button>
              <div class="create-account" onClick={this.toggleActive}>Create Account</div>
            </form>
            <form
              className="login-new-user"
              onSubmit={this.submitLogin}
              aria-label="Create new MovieTracker account"
            >
              <h2 className="new-user-h2">Create Account</h2>
              <label>
                <input
                  className="new-user-username"
                  tabIndex="1"
                  name="username"
                  placeholder="Username"
                  type="text"
                  value={username}
                  onChange={this.handleChange}
                />
              </label>
              <hr />
              <label>
                <input
                  className="new-user-email"
                  tabIndex="2"
                  name="email"
                  placeholder="Email"
                  type="email"
                  value={email}
                  onChange={this.handleChange}
                />
              </label>
              <hr />
              <label>
                <input
                  className="new-user-password"
                  tabIndex="3"
                  name="password"
                  placeholder="Password"
                  type="password"
                  value={password}
                  onChange={this.handleChange}
                />
              </label>
              <hr />
              <button tabIndex="1" className="new-user-submit">Submit</button>
            </form>
          </section>
        </section>
      </section>
    )
  }
}

export const mapDispatchToProps = dispatch => ({
  loginCheck: (user, password) => dispatch(loginUser(user, password)),
})

export default connect(
  null,
  mapDispatchToProps
)(Login)
