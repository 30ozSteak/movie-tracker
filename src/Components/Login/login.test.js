import React from 'react';
import ReactDOM from 'react-dom';
import {shallow, mount} from 'enzyme';
import configureStore from 'redux-mock-store';
import Login from './Login';


describe('Login', ()=>{

  const mockStore = configureStore();
  let wrapper;
  let loginCheck;
  let store;

  beforeEach(()=>{

    loginCheck = []

    store = mockStore(loginCheck);

    wrapper = shallow(<Login store={store}/>, store);

  })

  it('should match the snapshot', ()=>{
    expect(wrapper).toMatchSnapshot();
  })
})
