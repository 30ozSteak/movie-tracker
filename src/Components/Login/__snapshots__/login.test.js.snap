// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Login />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <section
          className="login-credentials"
        >
          <h2>
            Login
          </h2>
          <input
            className="user-name input-dimensions"
            placeholder="user name"
          />
          <input
            className="email input-dimensions"
            placeholder="email"
          />
          <Link
            replace={false}
            to="/main"
          >
            <button>
              Submit
            </button>
          </Link>
        </section>,
        <Route
          exact={true}
          path="/main"
          render={[Function]}
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h2>
              Login
            </h2>,
            <input
              className="user-name input-dimensions"
              placeholder="user name"
            />,
            <input
              className="email input-dimensions"
              placeholder="email"
            />,
            <Link
              replace={false}
              to="/main"
            >
              <button>
                Submit
              </button>
            </Link>,
          ],
          "className": "login-credentials",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Login",
            },
            "ref": null,
            "rendered": "Login",
            "type": "h2",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "user-name input-dimensions",
              "placeholder": "user name",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "email input-dimensions",
              "placeholder": "email",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <button>
                Submit
              </button>,
              "replace": false,
              "to": "/main",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Submit",
              },
              "ref": null,
              "rendered": "Submit",
              "type": "button",
            },
            "type": [Function],
          },
        ],
        "type": "section",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "exact": true,
          "path": "/main",
          "render": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <section
            className="login-credentials"
          >
            <h2>
              Login
            </h2>
            <input
              className="user-name input-dimensions"
              placeholder="user name"
            />
            <input
              className="email input-dimensions"
              placeholder="email"
            />
            <Link
              replace={false}
              to="/main"
            >
              <button>
                Submit
              </button>
            </Link>
          </section>,
          <Route
            exact={true}
            path="/main"
            render={[Function]}
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h2>
                Login
              </h2>,
              <input
                className="user-name input-dimensions"
                placeholder="user name"
              />,
              <input
                className="email input-dimensions"
                placeholder="email"
              />,
              <Link
                replace={false}
                to="/main"
              >
                <button>
                  Submit
                </button>
              </Link>,
            ],
            "className": "login-credentials",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Login",
              },
              "ref": null,
              "rendered": "Login",
              "type": "h2",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "user-name input-dimensions",
                "placeholder": "user name",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "email input-dimensions",
                "placeholder": "email",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <button>
                  Submit
                </button>,
                "replace": false,
                "to": "/main",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Submit",
                },
                "ref": null,
                "rendered": "Submit",
                "type": "button",
              },
              "type": [Function],
            },
          ],
          "type": "section",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "exact": true,
            "path": "/main",
            "render": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
